experimental/LieAlgebras/src/WeylGroup.jl

```diff
@@ -1,3 +1,189 @@
+@doc raw"""
+    exchange_left!(W::WeylGroup, w::AbstractVector{UInt8}, i::UInt8) -> AbstractVector{UInt8}
+
+Given a word `w` which is reduced w.r.t. `W`, modify `w` in-place into an equivalent word
+starting with `i` if the exchange condition can be applied, otherwise leave `w` unchanged.
+Finally, return `w`.
+
+!!! warning
+  If `w` is not a reduced decomposition, the behaviour is arbitrary.
```

Comment:
```suggestion
    If `w` is not a reduced expression, the behaviour is arbitrary.
```
this is how we call it a t other places